### PR TITLE
[Merged by Bors] - feat (RingTheory/PowerSeries) add formal derivative operation

### PR DESCRIPTION
**Pull request by rmhi**
Given a commutative semiring R, define the formal derivative R[[X]] \to R[[X]] and prove some basic properties.

This is a part of #7271.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grhkm21** [comment]
FYI you can link other PRs and issues with `#7271` -> #7271

**Comment by riccardobrasca** [comment]
Looking at polynomial we call the derivative as derivation [Polynomial.derivative'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/Derivation.html#Polynomial.derivative'). The `'` is an artifact of the fact that we have the unprimed version, but I think you should follow the same convention (i.e. remove the `f` from the name). I will have a closer look later today, but thanks!

File: Mathlib/RingTheory/PowerSeries/Derivative.lean [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  map_one_eq_zero'  := fDerivativeFun_one
+  leibniz'          := fDerivativeFun_mul
+/--Abbreviation of `PowerSeries.fDerivative`, the formal derivative on `R⟦X⟧`.-/
+scoped notation "d⁄dX" => fDerivative
+
+variable {R}
+
+@[simp] theorem fDerivative_C (r : R) : d⁄dX R (C R r) = 0 := fDerivativeFun_C r
+
+@[simp] theorem coeff_fDerivative (f : R⟦X⟧) (n : ℕ) :
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/PowerSeries/Derivative.lean:111]: Are you sure we want this as a `simp` lemma?

**Review by riccardobrasca — APPROVED** [review]
Beside my comment about the name LGTM, thanks!

bors d+

File: Mathlib/RingTheory/PowerSeries/Derivative.lean (hunk 1 above)
Comment (rmhi) [reply Mathlib/RingTheory/PowerSeries/Derivative.lean]: Thanks, you're right of course. I'm also making the name change that you suggest (i.e. removing the f).

**Pull request closed without merging**